Add gap-filling capability

astricklan opened the pull request:
Committed changes to both missing and advect_imf.  Added fill gaps to both DSCOVR and ACE.

astricklan committed:
Update missing.py

astricklan committed:
Update missing.py

astricklan committed:
Update missing.py

astricklan committed:
Update advect_imf.py

astricklan committed:
Update advect_imf.py

astricklan committed:
Update advect_imf.py

Added import missing and zero_to_function.  Converts -1e+31 bad data points to 9999999 for fill_gaps function in missing.py.

drsteve commented:
Thanks @astricklan - I might be missing it, but I can't see where the noise keyword argument gets passed in to the `load_acedata` or `load_dscovr` functions.

Also, can I suggest renaming `zero_to_nan` to something like `replace_fill` since it's basically not handling either zeros or NaNs at this point?

Final thing: can you check whether the `!=`s in `missing.py` (line 24) are what causes the issue with -1e31 as the fill value? You should be able to replace them with `~tb.feq(...)`

astricklan committed:
Update advect_imf.py

replaced zero_to_nan function with replace_fill.  Added noise argument to load_acedata and load_dscovr functions.

astricklan committed:
Update missing.py

Added tb.feq throughout first if statement.  Works fine now for -1e+31.

astricklan committed:
Update advect_imf.py

Removed replace_fill function.
